### ISSUE TITLE
Fallback chunk for production mode

### ISSUE THREAD

**Issue by joheriks** [vaadin/flow#6482]
Part of #6462 

1. Add a function in `generated-flow-imports.js` that dynamically imports another `generated-flow-imports-fallback.js` file, containing `import` statements for all dependencies on the classpath except those added to the original `generated-flow-imports.js` (in case the full bundle setting is used, the `generated-flow-imports-fallback.js` file can be generated empty). 

2. Add an entry to `flow-build-info.json` mapping each `@JSModule` and `@CssImport` to its bundle definition:
```
    chunks: {
        "fallback": { 
             jsModules: ["./mod1.js","./mod2.js", ... ], 
             cssImports: [ {filename: "styles.css", ... }, ... ] }
    }
```
Use current structure of `@CssImport` for objects inside the `cssImports` array.

3. UIDL metadata should inform client if it needs to load the fallback chunk before further processing. Current logic for `@HTMLImport` (and `@StyleSheet`, `@JavaScript`, ...) could be extended to do this as well. To detect whether the function should be triggered on the server side, use the same logic as in `DevModeInitializer.ensureAllDependenciesVisited` for dev mode. Architecture-wise, work same way as `Page.addHtmlImport`.

### Acceptance criteria ###
- [x] `generated-flow-imports-fallback.js` is generated as specified above
- [x] `generated-flow-imports-fallback.js` file is imported by webpack when required and only then
- [ ] IT that exercises the fallback chunk

**Comment by Legioth** [comment]
I would suggest that the dependency loading part would be implemented as a generic JS loader rather than hardcoding a dependency type specifically for this case. The idea would thus be that the server can send any JS expression as a "dependency" to load. That JS expression should return a Promise that resolves once the dependency is loaded.

In this way, the same mechanism can be used also for other cases such as RequireJS modules or manually defined webpack chunks as described in https://github.com/vaadin/flow/issues/5623#issuecomment-533093977. The key difference compared to `executeJs` is that updating the DOM and running `executeJs` would be postponed until all outstanding dependencies have been loaded.

**Comment by denis-anisimov** [comment]
This ticket is overcomplecated and contains some pieces which are not connected to each other at all. Also they are not even explicitly mentioned in the description.
So I will describe all parts of this big ticket here (I don't want to split it to subtickets) and will create several PRs one after another (or independent) to be able to accomplish this task.

- Generate `generated-flow-imports-fallback.js` in addition to `generated-flow-imports.js`. The file should 
  - not contain any theme related lines since they will be in the main imports file (and they may be only in one place). The theme is calculated using the main scanner strategy but it might be it's not discovered. In this case fallback strategy should be asked for it. The resulting theme should be used to rewrite module URIs in both files.
  - should not contain any module/script/css which is already in main `generated-flow-imports.js`
  - should contain all other module/script/css from the classpath which is not discovered by the main scanner strategy
- Add a function into `generated-flow-imports.js` which dynamically imports `generated-flow-imports-fallback.js` (it should be added because it will be compiled). The function should return a promise (`return  import("./generated-flow-imports-fallback.js";`) so that it's possible to do something when it's completed.
- As a note: webpack will generate a bunch of chunk files like `vaadin-0-xxxxx.js` , `vaadin-3-yyy.js`, etc.). Those files will be loaded when the function from above will be called. To be able to make this happen webpack should be able to load these files from the correct location. So it should be configured to look `VAADIN` folder for them. This requires additional config via `output: { publicPath: '/VAADIN/' }`
- add `chunks` object to `flow-build-info.json`(see the initial description). This file is the only available data in production mode since the scanner is not executed during production mode at all.
- forget about `DevModeInitializer.ensureAllDependenciesVisited`. As said in the previous item: we don't know anything about scanner and as a consequence we don't know which class is visited or not. `DevModeInitializer.ensureAllDependenciesVisited` logic works _ONLY_ in dev mode and the main purpose of this ticket to make it work in production mode.
- In fact logic related to `DevModeInitializer.ensureAllDependenciesVisited` should be just removed (but see the end of this list for that).
- `ComponentMetaData`  is used to collect data about dependencies per component class. It doesn't collect info for `CssImport`. It needs to be extended to collect this data. So it should discover all `CssImport`  annotations and put them into `DependencyInfo`. 
- The `DependencyInfo` instances from the previous step are used inside UIDL writer to add them dynamically via `Page`  instance inside `UIInternals::addComponentDependencies`. This method really adds all dependencies in compatibility mode (which are filtered then on the client side to be able to use a proper fragment). But it doesn't do anything for dependencies in NPM mode except external dependencies. Only external dependencies are loaded via `page.addJsModule`(or another method). As a result normal dependencies are just silently ignored. 
- Read `flow-build-info.json` inside `UIInternals::addComponentDependencies` somehow. (It might be that this file has to be read once in `DeploymentConfigurationFactory::readBuildInfo` where it's already read because it's location depends on init parameters. Then there is a question how to make available its data in `UIInternals::addComponentDependencies`). May be its data should be stored as a deployment configuration.
- Check whether the dependency is inside `fallback` chunk (data from `flow-build-info.json`). The logic should include the check for:
  - js module
  - jsscript
  - css imports
- In case if the dependency is inside `fallback` chunk we need to let the client side know about this (only once, so there should be a flag which says that the chunk has been already loaded).
- To be able to load the chunk we need to execute JS function which has been added into `generated-flow-imports.js` . Theoretically it may be done using `executeJs`. But it can't be done because JS is executed in the very end of message processing on the client side. But we want to execute this function before anything else.
- Add `Page:loadDependency` (or use some other method name)  which accepts a JS String and has semantic similar to `executeJS` : it should be possible to execute JS when dependency is loaded (assuming JS argument returns a `Promise`). It will require modifications on the server side, specific dependency type, modifications on the client side. This is standalone task which is not related to this ticket and may be done separately.
- Use the methods added from above to load dependency via the function  added into `generated-flow-imports.js`.
  - The question is : the dependency will be loaded async. But we in fact should not handle any other message on the client side until loading is completed. So dependency should be loaded not async. 
   - (see above) : the execution should use `await` either right away in the generated function or it should be `await call_generated_function` as a JS in the `loadDependency`  method. But then one should take care about `Promise` as a return value. I would suggest to use `await`  in the generated method to reduce hardcoded JS on the server side.
- Remove `VisitedClasses` and everything related to this including `Builder.collectVisitedClasses` and `FrontendDependenciesScanner::getClasses`. It's not needed anymore since the logic which checks whether the dependency has been loaded or not is in totally different place: it works now for both production and dev mode and it does some action (request to load fallback) instead of log a warning.


Looks like that's it.
It's quite long list and currently I've already done everything about files generation .
I will make a PR for that and proceed to work on other parts in other PRs.
